### ISSUE TITLE
[maven] Refine plugin dependencies

### ISSUE THREAD

**Issue by aalmiray** [jreleaser/jreleaser#1524]
Encountered the following in a dotCMS/core build

```
Warning:  
Warning:  Plugin validation issues were detected in 1 plugin(s)
Warning:  
Warning:   * org.jreleaser:jreleaser-maven-plugin:1.8.0
Warning:    Declared at location(s):
Warning:     * unknown
Warning:    Used in module(s):
Warning:     * com.dotcms:dotcms-cli-parent:1.0.0-SNAPSHOT (pom.xml)
Warning:    Plugin issue(s):
Warning:     * Plugin should declare these Maven artifacts in `provided` scope: [org.apache.maven:maven-artifact:3.8.7, org.apache.maven:maven-plugin-api:3.8.7, org.apache.maven:maven-model:3.8.7]
Warning:  
Warning:  
Warning:  Fix reported issues by adjusting plugin configuration or by upgrading above listed plugins. If no upgrade available, please notify plugin maintainers about reported issues.
Warning:  For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
Warning:  
```

https://github.com/dotCMS/core/actions/runs/6535055892/job/17743732677

**Comment by yrashish** [comment]
Can I take this up?

**Comment by yrashish** [comment]
I can't view action logs. "The logs for this run have expired and are no longer available." Does the fix require us to update the maven version [here](https://github.com/jreleaser/jreleaser/blob/v1.8.0/gradle.properties) in jreleaser version 1.8.0? If the warning appears to be related to maven version, then can we ask them to simply update the library as I could see that the latest version uses 3.8.8 of Maven.

**Comment by aalmiray** [comment]
Steps to reproduce:
1. clone https://github.com/jreleaser/helloworld-java-bin
2. edit `pom.xml` by adding the following plugins to the `<build>` section:
```xml
      <plugin>
          <groupId>org.jreleaser</groupId>
          <artifactId>jreleaser-maven-plugin</artifactId>
          <version>1.10.0</version>
      </plugin>
      <plugin>
          <groupId>org.jreleaser</groupId>
          <artifactId>jdks-maven-plugin</artifactId>
          <version>1.10.0</version>
      </plugin>
```
3. upgrade wrapper to Maven `3.9.6`
```
./mvnw wrapper:wrapper -DmavenVersion=3.9.6
```
4. Invoke the following command
```
./mvnw -Dmaven.plugin.validation=VERBOSE jreleaser:config jdks:setup-disco
```

Observe the warnings

```
[WARNING] 
[WARNING] Plugin [INTERNAL, EXTERNAL] validation issues were detected in following plugin(s)
[WARNING] 
[WARNING]  * org.jreleaser:jreleaser-maven-plugin:1.10.0
[WARNING]   Plugin EXTERNAL issue(s):
[WARNING]    * Plugin should declare Maven artifacts in `provided` scope. If the plugin already declares them in `provided` scope, update the maven-plugin-plugin to latest version. Artifacts found with wrong scope: [org.apache.maven:maven-plugin-api:3.8.8]
[WARNING] 
[WARNING]  * org.jreleaser:jdks-maven-plugin:1.10.0
[WARNING]   Plugin EXTERNAL issue(s):
[WARNING]    * Plugin should declare Maven artifacts in `provided` scope. If the plugin already declares them in `provided` scope, update the maven-plugin-plugin to latest version. Artifacts found with wrong scope: [org.apache.maven:maven-plugin-api:3.8.8]
[WARNING] 
[WARNING] 
[WARNING] Fix reported issues by adjusting plugin configuration or by upgrading above listed plugins. If no upgrade available, please notify plugin maintainers about reported issues.
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [NONE, INLINE, SUMMARY, BRIEF, VERBOSE]
[WARNING] 
```
6.

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.11.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.11.0))
